nmoscommon/mdns/mdnsRegistration.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python
+
+# Copyright 2017 British Broadcasting Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from zeroconf import ServiceInfo
+from socket import inet_aton
+
+
+class MDNSRegistration(object):
+
+    def __init__(self, interfaces, name, regtype, port, txtRecord=None):
+        if not txtRecord:
+            txtRecord = {}
+        self.name = name.replace(".", "-")
+        self.regtype = regtype
+        self.port = port
+        self.txtRecord = txtRecord
+        self.interfaces = interfaces
+        self.info = {}
+
+    def register(self):
+        nameList = self._makeNamesUnique()
+        regtype = self.regtype + ".local."
+        for interface in self.interfaces:
+            name = nameList[interface.ip] + "." + self.regtype + ".local."
+            self.info[interface.ip] = ServiceInfo(
+                type_=regtype,
+                name=name,
+                port=self.port,
+                properties=self.txtRecord,
+                address=inet_aton(interface.ip)
+            )
+            interface.registerService(self.info[interface.ip])
+
+    def _makeNamesUnique(self):
+        if len(self.interfaces) == 1:
```

Comment:
Any reason for the two ways of doing things in this method?

Reply:
As discussed - it's an attempt to retain backward compatibility for libraries that already provide the ip in "name", while still giving the multi-interface functionality we want going forward. If this ever becomes an issue, then there is an option of adding code in here to check to see if the name contains an IP already, but that feels a tad messy...